lib/Adapter/WorseReflection/Refactor/WorseGenerateAccessor.php

```diff
@@ -70,7 +62,23 @@ private function getInfo(SourceCode $sourceCode, int $offset): SymbolContext
             ));
         }
 
-        return $info;
+        return $this->generate($info, $sourceCode);
+    }
+
+    public function generateFromPropertyName(SourceCode $sourceCode, string $propertyName): SourceCode
+    {
+        $class = $this->class((string) $sourceCode);
+        $property = $class->properties()->get($propertyName);
+
+        $info = SymbolContext::for(Symbol::fromTypeNameAndPosition(
+            Symbol::PROPERTY,
+            $property->name(),
+            $property->position()
+        ))->withContainerType(Type::class($class->name()))
+        ->withTypes($property->docblock()->vars()->types())
+        ;
+
+        return $this->generate($info, $sourceCode);
```

Comment:
Instead of using the SymbolContext, maybe change the `generate` method definition to `(Type $containerType, string $propertyName, Type $propertyType)` ? the Symbol Context shouldn't really be part of the public WR API.

Reply:
I even wanted to gave it only a `ReflectionProperty` since it have all the needed information.

For the namespace and class name `$property->class()->name()` and `$property->class()->name()->namespace()`.

But to respect the law of demeter I propose to create:
* `ReflectionClassLike::namespace()`
* `ClassMemberReflection::namespace()`
* `ClassMemberReflection::className()`

Reply:
I guess we discussed this below, so will resolve this for now.